Sources/SwiftFormat/Core/SyntaxTraits.swift

```diff
@@ -65,3 +65,81 @@ extension ExprSyntax {
     return self.asProtocol(KeywordModifiedExprSyntaxProtocol.self) != nil
   }
 }
+
+/// Common protocol implemented by comma-separated lists whose elements
+/// support a `trailingComma`.
+protocol CommaSeparatedListSyntaxProtocol: SyntaxCollection where Element: WithTrailingCommaSyntax & Equatable {
+  /// Indicates whether the list should be treated as a collection for formatting.
+  /// If `true`, the list is affected by the `multiElementCollectionTrailingCommas` configuration.
+  var isCollection: Bool { get }
```

Comment:
Hate to be nitpicky, but can you change this to `isCollectionLiteral`? `isCollection` feels like it could be talking about the node itself (which are all `Collection`s).

Or... does this have to be a protocol requirement at all? It looks like you could just pass it in as another parameter to `markCommaDelimitedRegion`, since it has to be called individually for each node type anyway. That might be cleaner (but the rename is still valid elsewhere that it's referenced in the pretty printer).